plugins/modules/win_domain.ps1

```diff
@@ -9,13 +9,26 @@ $ErrorActionPreference = "Stop"
 
 # FUTURE: Consider action wrapper to manage reboots and credential changes
 
+$required_features = @("AD-Domain-Services","RSAT-ADDS", "RSAT-AD-AdminCenter")
+
+Function Get-MissingFeatures {
+    $features = @(Get-WindowsFeature $required_features)
+
+    If($features.Count -ne $required_features.Count) {
+        Throw "One or more Windows features required for a domain controller are unavailable"
+    }
+
+    $missing_features = @($features | Where-Object InstallState -ne Installed)
+
+    return ,$missing_features # no, the comma's not a typo- allows us to return an empty array
+}
+
 Function Install-Prereqs {
-    $gwf = Get-WindowsFeature AD-Domain-Services
-    if ($gwf.InstallState -ne "Installed") {
+    $missing_features = Get-MissingFeatures
+    if ($missing_features.Count -gt 0) {
```

Comment:
You can do this instead.

```suggestion
    if ($missing_features) {
```

By doing this you don't need the unary comma when returning/outputting the missing features in `Get-MissingFeatures` as the if statement will be true if it's a string or array. You could also just call `$missing_features = @(Get-MissingFeatures) to ensure the output is an array as well but that shouldn't be necessary.

Reply:
I find that things are more maintainable when object types don't change. Anyway, I've changed the condition as requested.